force-app/main/default/classes/TaskService.cls

```diff
@@ -4,4 +4,46 @@ public with sharing class TaskService
     {
 
     }
+
+    /**************************************************************
+    *   Creates a task associated with the opportunity 
+    *   to follow up in 10 days time 
+    *   Called in the context of an after insert trigger
```

Comment:
I'd argue that as it is a service method it should be able to be called from any context.